### PR TITLE
[CI] try to fix ci

### PR DESCRIPTION
**Pull request by SbloodyS**
## Purpose of the pull request

1. change path-filter logic to not-ignore to avoid contains docs path and not contains docs path bug.
2. change result logic.

### CODE REVIEW

File: .github/workflows/backend.yml [hunk 1]
```diff
@@ -81,13 +81,15 @@ jobs:
     name: Build
     runs-on: ubuntu-latest
     timeout-minutes: 30
-    needs: [ build ]
+    needs: [ build, paths-filter ]
```
Comment (kezhenxu94) [inline .github/workflows/backend.yml:84]: As [I said](https://github.com/apache/dolphinscheduler/pull/9325#discussion_r843403142) we don't need this?

File: .github/workflows/backend.yml [hunk 2]
```diff
@@ -81,13 +81,15 @@ jobs:
...
       - name: Status
         run: |
-          if [[ ${{ needs.build.result }} == 'success' || ${{ needs.paths-filter.outputs.ignore == 'true' }} ]]; then
+          if [[ ${{ needs.build.result }} == 'success' ]]; then
             echo "Passed!"
+          elif [[ ${{ needs.paths-filter.outputs.not-ignore }} == 'false' ]]; then
+            echo "Skip Build!"
           else
             exit -1
           fi
```
Comment (kezhenxu94) [inline .github/workflows/backend.yml]: Do you think this 👇 would be easier to understand?

```suggestion
          if [[ ${{ needs.paths-filter.outputs.ignore }} == 'true' ]]; then
            echo "Skip Build!"
            exit 0
          fi
          if [[ ${{ needs.build.result }} != 'success' ]]; then
            echo "Failed to build"
            exit -1
          fi
          # If we need to add more condition in the future, we can simply add
          if [[ ${{ needs.new.condition.result }} != 'success' ]]; then
            echo "Failed to do some new stuff"
            exit -1
          fi
```

File: .github/workflows/backend.yml (hunk 1 above)
Comment (SbloodyS) [reply .github/workflows/backend.yml:84]: It is not recognized in ```needs.paths-filter.outputs.not-ignore``` if not add this. Is there a better way to do that?

File: .github/workflows/backend.yml (hunk 2 above)
Comment (SbloodyS) [reply .github/workflows/backend.yml]: It's better. I'll fix it.

File: .github/workflows/backend.yml (hunk 1 above)
Comment (kezhenxu94) [reply .github/workflows/backend.yml:84]: > It is not recognized in `needs.paths-filter.outputs.not-ignore` if not add this. Is there a better way to do that?

OK

Comment (zhongjiajie) [reply .github/workflows/backend.yml:84]: So, do we still need this in `needs`?

Comment (kezhenxu94) [reply .github/workflows/backend.yml:84]: > So, do we still need this in `needs`?

Yes, in order to use the `paths-filter.output` in the `if` condition

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**